src/main/kotlin/no/nav/familie/ks/sak/kjerne/brev/BrevPeriodeContext.kt

```diff
@@ -276,20 +278,34 @@ class BrevPeriodeContext(
                 val nullableSanitybegrunnelse = vedtakBegrunnelse.begrunnelse.tilSanityBegrunnelse(sanityBegrunnelser)
                 nullableSanitybegrunnelse?.let { Pair(vedtakBegrunnelse.begrunnelse, it) }
             }.map { (begrunnelse, sanityBegrunnelse) ->
-                val personerMedVilkårSomPasserBegrunnelse = begrunnelserForPeriodeContext
-                    .hentPersonerMedVilkårResultaterSomPasserMedBegrunnelseOgPeriode(
+
+                val relevantePersoner = when (begrunnelse.begrunnelseType) {
+                    BegrunnelseType.ETTER_ENDRET_UTBETALING -> begrunnelserForPeriodeContext.hentPersonerMedEndretUtbetalingerSomPasserMedVedtaksperiode(
+                        sanityBegrunnelse
+                    )
+
+                    else -> begrunnelserForPeriodeContext.hentPersonerMedVilkårResultaterSomPasserMedBegrunnelseOgPeriode(
                         begrunnelse = begrunnelse,
                         sanityBegrunnelse = sanityBegrunnelse
                     )
+                }
 
                 val antallTimerBarnehageplass =
-                    hentAntallTimerBarnehageplassTekst(personerMedVilkårSomPasserBegrunnelse)
+                    hentAntallTimerBarnehageplassTekst(relevantePersoner)
+
+                val endringsperioder = begrunnelse.hentRelevanteEndringsperioderForBegrunnelse(
```

Comment:
Kan denne hete `relevanteEndringsperioderForBegrunnelse ` om ikke er det lett å tro at de gjelder for hele behandlingen

Reply:
Enig! 
Bør være 1-1 mellom variabelnavn og metodenavn her 👍